Cabal/Distribution/Simple/Register.hs

```diff
@@ -412,8 +412,9 @@ generalInstalledPackageInfo adjustRelIncDirs pkg abi_hash lib lbi clbi installDi
     -- Note. the libsubdir and datasubdir templates have already been expanded
     -- into libdir and datadir.
     IPI.libraryDirs        = if hasLibrary
-                               then libdir installDirs : extraLibDirs bi
-                               else                      extraLibDirs bi,
+                               then libdir installDirs    :
+                                    dynlibdir installDirs : extraLibDirs bi
```

Comment:
This bit will need to be extended to set the `libraryDynDirs` if that's supported by the target compiler, and otherwise put both libdir and dynlibdir in the `libraryDirs` as the patch does now.